Source/WebCore/html/shadow/TextControlInnerElements.cpp

```diff
@@ -254,13 +260,14 @@ std::optional<Style::ResolvedStyle> SearchFieldResultsButtonElement::resolveCust
     if (!shadowHostStyle)
         return std::nullopt;
 
-    auto appearance = shadowHostStyle->effectiveAppearance();
-    if (appearance == StyleAppearance::TextField) {
+    if (searchFieldStyleHasExplicitlySpecifiedTextFieldAppearance(*shadowHostStyle)) {
         auto elementStyle = resolveStyle(resolutionContext);
         elementStyle.style->setDisplay(DisplayType::None);
         return elementStyle;
     }
-    if (appearance != StyleAppearance::SearchField) {
+
+    auto effectiveAppearance = shadowHostStyle->effectiveAppearance();
+    if (effectiveAppearance != StyleAppearance::SearchField && effectiveAppearance != StyleAppearance::TextField) {
```

Comment:
Would be good to add a comment like:
// By default, input[type=search] can use either the searchfield or textfield appearance depending on the platform and writing mode. Only adjust the style when that default is used.

Reply:
Added.